keda/assets/logs/keda.yaml

```
@@ -0,0 +1,61 @@
+id: keda
+metric_id: keda
+backend_only: false
+installation_sources:
+  - keda
+  - keda-admission-webhooks
+  - keda-metrics-apiserver
+facets:
+pipeline:
+  type: pipeline
+  name: Keda
+  enabled: true
+  filter:
+    query: source:(keda OR keda-admission-webhooks OR keda-metrics-apiserver)
+  processors:
+    - type: grok-parser
+      name:  Keda Parser
+      enabled: true
+      source: message
+      samples:
+        - 'I1216 14:53:06.261024       1 main.go:112] "msg"="Connecting Metrics Service gRPC client to the server" "address"="keda-operator.keda.svc.cluster.local:9666" "logger"="keda_metrics_adapter"'
+        - 'I1216 14:53:06.680302       1 tlsconfig.go:243] "Starting DynamicServingCertificateController"'
+        - '2024-12-23T14:21:51Z	ERROR	scale_handler	error resolving auth params	{"type": "ScaledJob", "namespace": "default", "name": "csv-processor", "triggerIndex": 0, "error": "connection to redis failed: dial tcp 10.244.0.6:6379: connect: no route to host"}'
+        - '2024-12-23T14:21:38Z	INFO	KubeAPIWarningLogger	unknown field "status.authenticationsTypes"'
+        - '2024-12-16T14:53:12Z	INFO	controller-runtime.builder	Registering a validating webhook	{"GVK": "keda.sh/v1alpha1, Kind=ScaledObject", "path": "/validate-keda-sh-v1alpha1-scaledobject"}'
+      grok:
+        matchRules: |
+          keda_metrics_adapter3 %{_level_adapter}%{_date_adapter}\s+%{_pid}.*\s+%{_logger_adapter}]\s+"msg"="%{_msg}" %{data:component:keyvalue}
+          keda_metrics_adapter2 %{_level_adapter}%{_date_adapter}\s+%{_pid}.*\s+%{_logger_adapter}]\s+%{_msg}
+          keda_metrics_adapter1 %{_level_adapter}%{_date_adapter}\s+%{_pid}.*\s+%{_logger_adapter}]\s+"%{data::keyvalue}" %{data:component.name:keyvalue}
```

```suggestion
          keda_metrics_adapter1 %{_level_adapter}%{_date_adapter}\s+%{_pid}\s+%{_logger_adapter}]\s+"%{regex("[^\"]*")::keyvalue}" %{data:component.name:keyvalue}
```
- Not sure if there is a reason for .*?
- If we don't care about the content of "%{data::keyvalue}", can we just skip over it and match on the closing quote?

---

I can't really get this one working. Since it no longer parses:

```
I1216 14:53:06.261024 1 main.go:112] "msg"="Connecting Metrics Service gRPC client to the server" "address"="keda-operator.keda.svc.cluster.local:9666" "logger"="keda_metrics_adapter"
```

to:

```
{
  "date": 1734360786261,
  "msg": "\"Connecting Metrics Service gRPC client to the server\"",
  "process": {
    "id": 1
  },
  "address": "keda-operator.keda.svc.cluster.local:9666",
  "level": "I",
  "logger": "keda_metrics_adapter"
}
```

---

Actually this one is fine. I'll leave a new entry in the bottom. I had them numbered differently in the UI when I first created these rules.

---

I do apologize for the confusion here as there were more entries that I test than UI examples slots. I ended up reformatting that rule for a bit and moved it up since it's more specific. 

```
keda_metrics_adapter2 %{_level_adapter}%{_date_adapter}\s+%{_pid}\s+%{_logger_adapter}]\s+%{doubleQuotedString:msg} %{data:component.name:keyvalue}
```

For:

```
I1216 14:53:06.680001       1 dynamic_serving_content.go:135] "Starting controller" name="serving-cert::/certs/tls.crt::/certs/tls.key"
```

---

Makes sense to me!